### PR TITLE
create a test for quantized resnet50 - extended

### PR DESCRIPTION
**Pull request by wojtuss**
This is an extended version of https://github.com/PaddlePaddle/Paddle/pull/16399.
It supports inference on a full ImageNet dataset preprocessed using the tool https://github.com/PaddlePaddle/Paddle/pull/16529

With the use of this test we get the following accuracy results on the preprocessed full imagenet val dataset:
ResNet50:
Avg top1 INT8 accuracy: 0.7640
Avg top1 FP32 accuracy: 0.7663
MobileNet-v1:
Avg top1 INT8 accuracy: 0.7039
Avg top1 FP32 accuracy: 0.7078

### CODE REVIEW

**Comment by sfraczek** [comment]
It also contains bug fixes and other review fixes.

File: paddle/fluid/inference/tests/api/analyzer_int8_image_classification_tester.cc [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+void SetConfig(AnalysisConfig *cfg) {
+  cfg->SetModel(FLAGS_infer_model);
+  cfg->SetProgFile("__model__");
+  cfg->DisableGpu();
+  cfg->SwitchIrOptim();
+  cfg->SwitchSpecifyInputNames(false);
+  cfg->SetCpuMathLibraryNumThreads(FLAGS_paddle_num_threads);
+  cfg->EnableMKLDNN();
+  cfg->pass_builder()->SetPasses(
```
Comment (sfraczek) [inline paddle/fluid/inference/tests/api/analyzer_int8_image_classification_tester.cc:34]: @luotao1 without this function it didn't work right due to bug with fuse passes being copied many times.

**Comment by wojtuss** [comment]
@luotao1, this version requires a dataset with a different name on the server. @lidanqing-intel will send you the data.

File: paddle/fluid/inference/tests/api/analyzer_int8_image_classification_tester.cc (hunk 1 above)
Comment (luotao1) [reply paddle/fluid/inference/tests/api/analyzer_int8_image_classification_tester.cc:34]: Do you mean the same bug in #16174?

**Comment by luotao1** [comment]
Is #16399 closed due to this PR?

**Comment by luotao1** [comment]
I upload the data and rerun the CI.

**Comment by luotao1** [comment]
```
[02:21:05]        Start 164: test_analyzer_int8_resnet50
[02:21:06]147/610 Test #164: test_analyzer_int8_resnet50 .....................***Exception: SegFault  0.40 sec
[02:21:06]WARNING: Logging before InitGoogleLogging() is written to STDERR
[02:21:06]W0329 02:21:05.684131 57099 init.cc:83] Cannot enable P2P access from 0 to 1
[02:21:06]W0329 02:21:05.684239 57099 init.cc:83] Cannot enable P2P access from 1 to 0
[02:21:06][==========] Running 2 tests from 1 test case.
[02:21:06][----------] Global test environment set-up.
[02:21:06][----------] 2 tests from Analyzer_int8_resnet50
[02:21:06][ RUN      ] Analyzer_int8_resnet50.quantization
[02:21:06]/paddle/paddle/fluid/inference/tests/api/analyzer_int8_image_classification_tester.cc:122: Failure
[02:21:06]Failed
[02:21:06]Couldn't open file: /root/.cache/inference_demo/int8/data.bin
[02:21:06]
[02:21:06]        Start 165: test_analyzer_int8_mobilenet
[02:21:06]148/610 Test #165: test_analyzer_int8_mobilenet ....................***Exception: SegFault  0.40 sec
[02:21:06]WARNING: Logging before InitGoogleLogging() is written to STDERR
[02:21:06]W0329 02:21:06.362413 57101 init.cc:83] Cannot enable P2P access from 0 to 1
[02:21:06]W0329 02:21:06.362510 57101 init.cc:83] Cannot enable P2P access from 1 to 0
[02:21:06][==========] Running 2 tests from 1 test case.
[02:21:06][----------] Global test environment set-up.
[02:21:06][----------] 2 tests from Analyzer_int8_resnet50
[02:21:06][ RUN      ] Analyzer_int8_resnet50.quantization
[02:21:06]/paddle/paddle/fluid/inference/tests/api/analyzer_int8_image_classification_tester.cc:122: Failure
[02:21:06]Failed
[02:21:06]Couldn't open file: /root/.cache/inference_demo/int8/data.bin
[02:21:06]
```

**Pull request closed without merging**